cppcache/src/ThinClientPoolDM.cpp

```diff
@@ -703,42 +703,36 @@ GfErrType ThinClientPoolDM::sendRequestToAllServers(
 
 const std::shared_ptr<CacheableStringArray> ThinClientPoolDM::getLocators()
     const {
-  auto ptrArr =
-      new std::shared_ptr<CacheableString>[m_attrs->m_initLocList.size()];
-  int32_t i = 0;
+  std::vector<std::shared_ptr<CacheableString>> locators(
+      m_attrs->m_initLocList.size());
+  uint32_t i = 0;
   for (auto&& locator : m_attrs->m_initLocList) {
-    ptrArr[i++] = CacheableString::create(locator);
+    locators[i++] = CacheableString::create(locator);
```

Comment:
Can you just call push_back here, like you're doing below for getServers(), and eliminate the variable `i`?

Reply:
Yes. I will.